### PR TITLE
Validation fixed for register tree issue #1577

### LINKED ISSUE #1577: Register trees: Don't show errors for valid input if other input is missing
**Describe the bug**
If you do not specify a location for registering a tree, but already entered a valid date, next to the error for specifying a location also an error for the date is shown.

**Is it on a Production or Development Build?**
production and develop

**To Reproduce**
1. Go to 'Register trees'
2. Enter a valid date
3. Click button to register

**Expected behavior**
Just show an error for the missing location, but not for the date

**What actually happens**
Showing errors for the date and the location.

**Screenshots**
Bildschirmfoto 2019-10-30 um 12 04 40

**Desktop (please complete the following information):**
- OS: latest MacOS
- Browser latest Chrome
- Plant-for-the-Planet App Version 1.1.0

### PR DESCRIPTION
**Pull request by rabiul042**
I have fixed the form validation problem and parameter passing issue through the API of single and multiple trees.

Fix #1577

### CODE REVIEW

File: app/components/RegisterTrees/index.js [hunk 1]
```diff
@@ -139,7 +139,12 @@ export default class RegisterTrees extends Component {
...
     }
 
-    this.props.onSubmit(this.state.mode, null, plantProject);
+    let formvalue =
+      this.state.mode == 'single-tree'
+        ? this.state.formValueSingle
+        : this.state.formValueMultiple;
+
+    this.props.onSubmit(this.state.mode, formvalue, plantProject);
     event.preventDefault();
```
Comment (mahmed0715) [inline app/components/RegisterTrees/index.js]: @rabiul042 React never allows you to directly set a params in this.state.
You have to use this.setState() to update any state params...
Please check react docs

**Comment by rabiul042** [comment]
Okay, Thanks

File: app/containers/RegisterTrees/index.js [hunk 2]
```diff
@@ -27,12 +27,18 @@ class RegisterTreesContainer extends PureComponent {
...
   onSubmit = (mode, registerTreeForm, plantProject) => {
-    registerTreeForm =
+    /*registerTreeForm =
       registerTreeForm || this.refs.registerTrees.refs.registerTreeForm;
     // console.log(registerTreeForm.validate());
     let value = registerTreeForm.getValue();
     value = Object.assign({}, value);
+    value = mergeContributionImages(value);*/
+
+    let result = this.refs.registerTrees.refs.registerTreeForm.validate();
```
Comment (mahmed0715) [inline app/containers/RegisterTrees/index.js]: Please run 
```npm run lint:errors```
and fix eslint errors.

**Comment by mahmed0715** [comment]
@norbertschuler please  check this PR, is this the correct fix?

File: app/server/formSchemas/registerTrees.js [hunk 3]
```diff
@@ -130,7 +130,7 @@ export default {
         propertyOrder: 8
       }
     },
-    required: ['treeCount', 'plantDate', 'geoLocation'],
```
Comment (norbertschuler) [inline app/server/formSchemas/registerTrees.js]: Who has made the decision that geolocation is not a required input value for tree registration any more?

Comment (norbertschuler) [reply app/server/formSchemas/registerTrees.js]: Btw. (not part of this patch) if one does not enter a tree type name then the resulting entries at my forest shows "null" for it which does not look very good, but is another error:
Bildschirmfoto 2019-11-13 um 16 30 22
Also deleting a tree from my forest still does not update the client which again is another error #1100

**Comment by mahmed0715** [comment]
### @norbertschuler the api is not asking for geolocation anymore. I checked that.

**Comment by norbertschuler** [comment]
> @norbertschuler the api is not asking for geolocation anymore. I checked that. This field is nit required anymore.

Yes, I saw that. My point was just, do we really want to allow registrations of trees without location? @sagararyal

**Comment by sagararyal** [comment]
@norbertschuler @mahmed0715 

Geolocation should always be required. Else this will create issues in future.

File: app/server/formSchemas/registerTrees.js (hunk 3 above)
Comment (norbertschuler) [reply app/server/formSchemas/registerTrees.js]: So as Sagar said `geoLocation` is an mandatory field, let's keep it. If the backend does not throw an error if it’s missing then that’s a backend issue to be solved.

Comment (mahmed0715) [reply app/server/formSchemas/registerTrees.js]: Changed geolocation as mandatory.

**Comment by mahmed0715** [comment]
@norbertschuler I think its good now to be merged. @rabiul042 as first timer.

**Pull request merged**